build/Release.ps1

```diff
@@ -0,0 +1,156 @@
+    [bool]
+    $CoreComponents,
+    [Parameter(Mandatory)]
+    [ValidateNotNullOrEmpty()]
+    [bool]
+    $NonCoreComponents,
+    [Parameter(Mandatory)]
+    [ValidateNotNullOrEmpty()]
+    [bool]
+    $IsStableRelease
+)
+
+Function PressKeyToContinue {
+    param (
+        $Instruction
+    )
+
+    Write-Host $Instruction
+    Write-Host -NoNewLine "Press any key to continue..."
+    $Null = $Host.UI.RawUI.ReadKey("NoEcho,IncludeKeyDown")
+}
+
+Function GitFetchAndMergeUpstreamMain {
+    git fetch upstream main
+    git checkout main
+    git merge upstream/main
+}
+
+# Ensure branch is clean, otherwise abort
+$ChangedFiles = $(git status --porcelain | Measure-Object | Select-Object -Expand Count)
+if ($ChangedFiles -gt 0) {
+    Write-Error "Aborting. Found uncommitted changes." -ErrorAction Stop
+}
+
+# Checkout `main` branch and create new branch for changes
+$ChangelogBranch = "changelogs-$TagName"
+GitFetchAndMergeUpstreamMain
+git checkout -b $ChangelogBranch
+
+# CHANGELOG Processing and Updating
+$CombinedChangelog = ""
+ForEach ($ChangelogPath in Get-ChildItem -Path . -Recurse -Filter CHANGELOG.md) {
+    $CombinedChangelog += "**$($ChangelogPath.Directory.Name)**"
+    $ChangelogContent = Get-Content -Path $ChangelogPath.FullName
+    $IsStarted = $False
+    ForEach ($Line in $ChangelogContent) {
+        If (-not $IsStarted -and $Line -like "## Unreleased") {
+            $IsStarted = $True
+        }
+        elseif ($IsStarted -and $Line -like "## *") {
+            break
+        }
+        elseif ($IsStarted) {
+            $CombinedChangelog += $Line
+        }
+    }
+
+    $NewFileHeader = "Unreleased`n`n## $TagName`n`nReleased $(Get-Date -UFormat '%Y-%b-%d')"
+    (Get-Content -Path $ChangelogPath.FullName) -Replace "Unreleased", $NewFileHeader | Set-Content -Path $ChangelogPath.FullName
+}
+
+# Normalize PublicApi files
+If ($IsStableRelease) {
+    Start-Process PowerShell.exe -ArgumentList "-noexit", "-command .\build\finalize-publicapi.ps1"
+}
+
+# Submit PR and Merge
+$ChangelogTitle = "CHANGELOG.md Files Update - $TagName"
+git commit -am $ChangelogTitle
+git push --set-upstream origin $ChangelogBranch
+gh pr create --title $ChangelogTitle # TODO --reviewer @reyang
+PressKeyToContinue("Merge the PR")
+
+# Tag Git with version to be released
+GitFetchAndMergeUpstreamMain
+git tag -a $TagName -m $TagName
+If ($CoreComponents) {
+    git tag -a "core-$TagName" -m "$TagName of all core components"
+}
+If ($NonCoreComponents) {
+    git tag -a $TagName -m "$TagName of all non-core components"
+}
+git push origin $TagName
+
+# Trigger MyGet Build and Download Artifacts from the Drop
+$DownloadDirectory = "$HOME/Downloads"
+gh workflow run "publish-packages-1.0.yml"
+gh run download --name "windows-latest-packages" --dir $DownloadDirectory
+Push-Location "$DownloadDirectory/src"
+Invoke-WebRequest -Uri "https://dist.nuget.org/win-x86-commandline/latest/nuget.exe" -OutFile "src/nuget.exe"
```

Comment:
What if some command like this one fails?
It has been a long time since I have developed PS code, but I guess we can try using `ErrorAction` params or/and `ErrorActionPreference`